### PR TITLE
Doc:Guide to deploy Databricks on Azure with PrivateLink - Standard deployment

### PR DESCRIPTION
**Pull request by yessawab**
Added a new guide for the Standard deployment approach of Azure Databricks with Azure Private Link ([Doc](https://learn.microsoft.com/en-us/azure/databricks/administration-guide/cloud-configurations/azure/private-link-standard))

### CODE REVIEW

File: docs/guides/azure-private-link-workspace-standard.md [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+
+This guide uses the following variables:
+
+- `cidr_transit`: The CIDR for the Azure transit VNet
+- `cidr_app`: The CIDR for the Azure Data Plane VNet
+- `rg_transit`: The name of the existing resource group that will contain the Azure transit VNet and the private DNS zone for the Frontend and the Web auth private endpoint
+- `rg_app`: The name of the existing resource group that will contain the Azure Data Plane VNet and the private DNS zone for the Backend private endpoint
+- `location`: The location for Azure resources
+
+This guide is provided as-is and you can use it as the basis for your custom Terraform module.
```
Comment (alexott) [inline docs/guides/azure-private-link-workspace-standard.md:39]: Should we point to https://github.com/databricks/terraform-databricks-examples/tree/main/modules/adb-with-private-link-standard ?

File: docs/guides/azure-private-link-workspace-standard.md [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+
+To get started with Azure Private Link integration, this guide takes you through the following high-level steps:
+
+- Initialize the required providers
+- Configure Azure objects
+  - Deploy two Azure VNets with the following subnets:
+    - Public and private subnets for Azure Databricks workspace in the Data Plane VNet
```
Comment (alexott) [inline docs/guides/azure-private-link-workspace-standard.md:46]: if we provision two workspaces, can we make clear that we need separate subnets for each of them?

File: docs/guides/azure-private-link-workspace-standard.md [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+## Provider initialization
+
+Initialize provider 
+
+```hcl
+terraform {
+  required_providers {
+    databricks = {
+      source = "databricks/databricks"
+    }
```
Comment (alexott) [inline docs/guides/azure-private-link-workspace-standard.md:65]: We don't need it in this guide

**Review by alexott — CHANGES_REQUESTED** [review]
it would be nice to address small comments

**Comment by nfx** [comment]
Following up - is this issue still relevant?

**Comment by yessawab** [comment]
> Following up - is this issue still relevant?

I made the requested changes. @alexott can approve it.

File: docs/guides/azure-private-link-workspace-simplified.md [hunk 4]
```diff
@@ -2,23 +2,24 @@
 page_title: "Provisioning Databricks on Azure with Private Link - Simple deployment"
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-simplified.md:2]:
```suggestion
page_title: "Provisioning Azure Databricks with Private Link - Simple deployment"
```

File: docs/guides/azure-private-link-workspace-standard.md [hunk 5]
```diff
@@ -0,0 +1,512 @@
+---
+page_title: "Provisioning Databricks on Azure with Private Link - Standard deployment"
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-standard.md:2]:
```suggestion
page_title: "Provisioning Azure Databricks with Private Link - Standard deployment"
```

File: docs/guides/azure-private-link-workspace-standard.md [hunk 6]
```diff
@@ -0,0 +1,512 @@
...
+---
+
+# Deploying pre-requisite resources and enabling Private Link connections - Standard deployment
+
+-> **Note** 
+  - Refer to [adb-with-private-link-standard](https://github.com/databricks/terraform-databricks-examples/tree/main/modules/adb-with-private-link-standard), a Terraform module that contains code used to deploy an Azure Databricks workspace with Azure Private Link using the Standard deployment approach. 
+  - Refer to the [Databricks Terraform Registry modules](https://registry.terraform.io/modules/databricks/examples/databricks/latest) for more Terraform modules and examples to deploy Azure Databricks resources.
+  - This guide assumes that connectivity from the on-premises user environment is already configured using ExpressRoute or a VPN gateway connection.
+
+Databricks Private Link support enables private connectivity between users and their Databricks workspaces and between clusters on the data plane and core services on the control plane within the Databricks workspace infrastructure.
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-standard.md:12]:
```suggestion
[Azure Private Link](https://learn.microsoft.com/en-us/azure/private-link/private-link-overview) support enables private connectivity between users and their Databricks workspaces and between clusters on the data plane and core services on the control plane within the Databricks workspace infrastructure. 
```

File: docs/guides/azure-private-link-workspace-simplified.md [hunk 7]
```diff
@@ -2,23 +2,24 @@
...
 ---
 
-# Deploying pre-requisite resources and enabling Private Link connections
+# Deploying pre-requisite resources and enabling Private Link connections - Simple deployment
+
+-> **Note** Refer to the [Databricks Terraform Registry modules](https://registry.terraform.io/modules/databricks/examples/databricks/latest) for Terraform modules and examples to deploy Azure Databricks resources.
 
 -> **Note** This guide assumes that connectivity from the on-premises user environment is already configured using ExpressRoute or a VPN gateway connection.
 
 Databricks Private Link support enables private connectivity between users and their Databricks workspaces and between clusters on the data plane and core services on the control plane within the Databricks workspace infrastructure.
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-simplified.md:11]:
```suggestion
[Azure Private Link](https://learn.microsoft.com/en-us/azure/private-link/private-link-overview) support enables private connectivity between users and their Databricks workspaces and between clusters on the data plane and core services on the control plane within the Databricks workspace infrastructure. 
```

File: docs/guides/azure-private-link-workspace-standard.md [hunk 8]
```diff
@@ -0,0 +1,512 @@
...
+
+data "azurerm_client_config" "current" {
+}
+
+data "external" "me" {
+  program = ["az", "account", "show", "--query", "user"]
+}
+
+locals {
+  prefix = "abd-pl"
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-standard.md:107]:
```suggestion
  prefix = "adb-pl"
```

File: docs/guides/azure-private-link-workspace-standard.md [hunk 9]
```diff
@@ -0,0 +1,512 @@
...
+
+  private_dns_zone_group {
+    name                 = "app-private-dns-zone-dpcp"
+    private_dns_zone_ids = [azurerm_private_dns_zone.dnsdpcp.id]
+  }
+}
+```
+
+-> **Note** 
+- The public network access to the workspace is disabled. You can access the workspace only through the private connectivity to the on-premises user environment. For testing purposes, you can deploy an Azure VM in the Transit vnet in order to test the frontend connectivity.
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-standard.md:511]:
```suggestion
- The public network access to the workspace is disabled. You can access the workspace only through the private connectivity to the on-premises user environment. For testing purposes, you can deploy an Azure VM in the Transit VNet in order to test the frontend connectivity.
```

File: docs/guides/azure-private-link-workspace-standard.md [hunk 10]
```diff
@@ -0,0 +1,512 @@
...
+  network_security_group_rules_required = "NoAzureDatabricksRules" //use private endpoint
+  customer_managed_key_enabled          = true
+  custom_parameters {
+    no_public_ip                                         = true
+    virtual_network_id                                   = azurerm_virtual_network.app_vnet.id
+    private_subnet_name                                  = azurerm_subnet.app_private.name
+    public_subnet_name                                   = azurerm_subnet.app_public.name
+    public_subnet_network_security_group_association_id  = azurerm_subnet_network_security_group_association.app_public.id
+    private_subnet_network_security_group_association_id = azurerm_subnet_network_security_group_association.app_private.id
+    storage_account_name                                 = "dbfsapp723k4b3"
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-standard.md:477]: should this be hardcoded?

File: docs/guides/azure-private-link-workspace-standard.md [hunk 11]
```diff
@@ -0,0 +1,512 @@
...
+  network_security_group_rules_required = "NoAzureDatabricksRules" //use private endpoint
+  customer_managed_key_enabled          = true
+  custom_parameters {
+    no_public_ip                                         = true
+    virtual_network_id                                   = azurerm_virtual_network.transit_vnet.id
+    private_subnet_name                                  = azurerm_subnet.transit_private.name
+    public_subnet_name                                   = azurerm_subnet.transit_public.name
+    public_subnet_network_security_group_association_id  = azurerm_subnet_network_security_group_association.transit_public.id
+    private_subnet_network_security_group_association_id = azurerm_subnet_network_security_group_association.transit_private.id
+    storage_account_name                                 = "dbfstransitd723k4b3"
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-standard.md:295]: hardcoded storage account name?

File: docs/guides/azure-private-link-workspace-standard.md [hunk 12]
```diff
@@ -0,0 +1,512 @@
...
+
+-> **Note**  
+* A seperate Web Auth workspace is not mandatory but recommended.
+* DNS mapping for SSO login callbacks to the Azure Databricks web application can either be managed by the Web Auth workspace or another workspace that is associated with the **browser_authentication** private endpoint.
+
+![Azure Databricks with Private Link - Standard deployment](https://github.com/databricks/terraform-provider-databricks/raw/master/docs/images/azure-private-link-standard.png)
+
+This guide uses the following variables:
+
+- `cidr_transit`: The network range (CIDR) for the Azure transit VNet
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-standard.md:34]: what are the minimum VNet size for these?

File: docs/guides/azure-private-link-workspace-standard.md [hunk 13]
```diff
@@ -0,0 +1,512 @@
...
+## Deploy Transit resources
+
+1. Create a Transit VNet
+
+```hcl
+resource "azurerm_virtual_network" "transit_vnet" {
+  name                = "${local.prefix}-transit-vnet"
+  location            = var.location
+  resource_group_name = var.rg_transit
+  address_space       = [local.cidr_transit]
```
Comment (nkvuong) [inline docs/guides/azure-private-link-workspace-standard.md:137]: could you double check all instance of `local` vs. `var` further down?

```suggestion
  address_space       = [var.cidr_transit]
```

**Review by nkvuong — CHANGES_REQUESTED** [review]
@yessawab A few minor changes from me please

File: docs/guides/azure-private-link-workspace-standard.md (hunk 11 above)
Comment (yessawab) [reply docs/guides/azure-private-link-workspace-standard.md]: Changed it using random_string resource

File: docs/guides/azure-private-link-workspace-standard.md (hunk 12 above)
Comment (yessawab) [reply docs/guides/azure-private-link-workspace-standard.md:34]: Normally a CIDR block between /16 and /24 for the VNets. 
But we deploy private endpoints and we deploy a workspace in both vnets. One of them is for browser_auth. 
I think /23 should be the minimum.

**Review by alexott — APPROVED** [review]
lgtm

**Pull request merged**